contract/contracts/Elections/ElectionUtils.sol

```diff
@@ -0,0 +1,230 @@
+pragma solidity ^0.8.24;
+
+/**
+ * @title ElectionUtils
+ * @dev Utility functions for election validation and calculations
+ */
+library ElectionUtils {
+  /**
+   * @dev Represents a rank-order vote submitted by a voter
+   */
+  struct RankOrderVote {
+    address voter;
+    address[] rankedCandidates; // Ordered from most preferred to least
+  }
+
+  /**
+   * @dev Candidate result with calculated points for sorting
+   */
+  struct CandidateResult {
+    address candidateAddress;
+    uint256 totalPoints;
+    uint256 rank;
+    bool isWinner;
+  }
+  /**
+   * @dev Error thrown when seat count is not odd
+   */
+  error InvalidSeatCount();
+
+  /**
+   * @dev Error thrown when date validation fails
+   */
+  error InvalidDates();
+
+  /**
+   * @dev Error thrown when there are insufficient candidates
+   */
+  error InsufficientCandidates();
+
+  /**
+   * @dev Error thrown when duplicate candidates are detected
+   */
+  error DuplicateCandidates();
+
+  /**
+   * @dev Error thrown when voter list is empty
+   */
+  error NoEligibleVoters();
+
+  /**
+   * @dev Error thrown when duplicate voters are detected
+   */
+  error DuplicateVoters();
+
+  uint256 private constant MAX_CANDIDATES = 20;
+
+  /**
+   * @dev Validates that seat count is odd
+   * @param seatCount Number of seats to validate
+   */
+  function validateSeatCount(uint256 seatCount) internal pure {
+    if (seatCount == 0 || seatCount % 2 == 0) {
+      revert InvalidSeatCount();
+    }
+  }
+
+  /**
+   * @dev Validates election dates
+   * @param startDate Election start timestamp
+   * @param endDate Election end timestamp
+   */
+  function validateDates(uint256 startDate, uint256 endDate) internal view {
+    if (startDate <= block.timestamp || endDate <= startDate) {
+      revert InvalidDates();
+    }
+  }
+
+  /**
+   * @dev Validates candidate list
+   * @param candidates Array of candidates
+   * @param seatCount Number of seats
+   */
+  function validateCandidates(address[] memory candidates, uint256 seatCount) internal pure {
+    if (candidates.length < seatCount) {
+      revert InsufficientCandidates();
+    }
+
+    // Check for duplicate candidate addresses
+    for (uint256 i = 0; i < candidates.length; i++) {
+      for (uint256 j = i + 1; j < candidates.length; j++) {
+        if (candidates[i] == candidates[j]) {
+          revert DuplicateCandidates();
+        }
+      }
+    }
+  }
+
+  /**
+   * @dev Validates eligible voters list
+   * @param eligibleVoters Array of voter addresses
+   */
+  function validateVoters(address[] memory eligibleVoters) internal pure {
+    if (eligibleVoters.length == 0) {
+      revert NoEligibleVoters();
+    }
+
+    // Check for duplicate voter addresses
+    for (uint256 i = 0; i < eligibleVoters.length; i++) {
+      for (uint256 j = i + 1; j < eligibleVoters.length; j++) {
+        if (eligibleVoters[i] == eligibleVoters[j]) {
+          revert DuplicateVoters();
+        }
+      }
```

Comment:
And this, can use one function that detect if there is a duplication in an array. 
And if there is a duplication you revert